lambda-otel4s/jvm/src/test/scala/feral/lambda/otel4s/TracedHandlerSuite.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral.lambda
+package otel4s
+
+import cats.effect.IO
+import cats.effect.kernel.Resource
+import cats.syntax.all._
+import com.amazonaws.services.lambda.runtime
+import io.circe.Decoder
+import io.circe.Encoder
+import io.circe.syntax._
+import munit.CatsEffectSuite
+import org.typelevel.otel4s.Attributes
+import org.typelevel.otel4s.sdk.testkit.trace.TracesTestkit
+import org.typelevel.otel4s.semconv.experimental.attributes.CloudExperimentalAttributes
+import org.typelevel.otel4s.semconv.experimental.attributes.CloudExperimentalAttributes.CloudProviderValue
+import org.typelevel.otel4s.semconv.experimental.attributes.FaasExperimentalAttributes
+import org.typelevel.otel4s.trace.SpanKind
+
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.util.concurrent.atomic.AtomicInteger
+
+class TracedHandlerSuite extends CatsEffectSuite {
```

Comment:
Can we avoid the substantial duplication/splitting of these suite? It looks like you based these on some tests that needed extra machinery because they were testing the platform-specific integration. But I think in your situation it should be possible to test `TracedHandler` using pure code.

Reply:
Sure thing, makes sense